test/conformance/runtime/readiness_probe_test.go

```diff
@@ -28,31 +28,61 @@ import (
 	v1a1test "knative.dev/serving/test/v1alpha1"
 )
 
+// testCases for table-driven testing.
+var testCases = []struct {
+	// name of the test case, which will be inserted in names of routes, configurations, etc.
+	// Use a short name here to avoid hitting the 63-character limit in names
+	// (e.g., "service-to-service-call-svc-cluster-local-uagkdshh-frkml-service" is too long.)
+	name string
+	// handler to be used for readiness probe in user container.
+	handler corev1.Handler
+}{
+	{
+		"httpGet",
+		corev1.Handler{
+			HTTPGet: &corev1.HTTPGetAction{
+				Path: "/healthz",
+			},
+		},
+	},
+	{
+		"exec",
+		corev1.Handler{
+			Exec: &corev1.ExecAction{
+				Command: []string{"/ko-app/runtime", "probe"},
+			},
```

Comment:
Can you add a case for TCP while you are in here?

I feel like for conformance we should have variants of these tests that FAIL the readiness probes.  Could you either add that or open an issue to follow up?

cc @dgerd

Reply:
We've sort of been here before, but the [runtime contract](https://github.com/knative/serving/blob/master/docs/runtime-contract.md#meta-requests) suggests that only `httpGet` and `tcpSocket` probes should be used. We've quietly left them in, but do we want to codify their allowance into our conformance tests?